### ISSUE TITLE
So What's the Status of the Domain System?

### ISSUE THREAD

**Issue by smreyno** [Arx-Game/arxcode#239]
Presently it feels like House elevation and House power in general is mostly a product of OOC/population-based resource grinding. Maybe this is fine and in that case ignore me! That's certainly a way to do things! But it was my understanding an actual domain system would work somewhat differently, with more mechanics built in for passive maintenance and clearer paths to increase certain aspects of a domain's power. Want to get a bead on this given all the House elevation going on atm (yes, the Pravus drama is the impetus of this, but it seems like this is a growing way to level up and I want to drop this note because idk /I/ care about this and always liked the idea of more concrete power-advancement avenues that weren't totally diffuse/grinding-oriented). I appreciate all y'all do, etc.

ETA: And if things that might logically prompt civil war are being allowed due to resource accumulation, it would be nice to have a system in place to account for such conflict! To get everyone on the same page ICly and OOCly. It seems they aren't at present. NGL the game's approach to PvP has always felt more diffuse than is healthy to my liking, but that didn't matter as long as pretty much all power stayed in the diffuse realm. It's plainly being treated as a more concrete thing now.

**Comment by TehomCD** [comment]
Yeah, I agree it'd be nice to have a system in place for it - right now it's pretty ad hoc, I believe. My recommendation would be probably brainstorming a robust rule system for it that we would automate later - if it's fully specced out, that'd at least cover the design work. I was a huge fan of 4X games back in the day but at this point I'd cheerfully implement any minimum viable product that people find at least somewhat reasonable.

I don't really have much time to design systems these days. Even ignoring my time being evaporated by other obligations, there's a huge laundry list of technical requirements that are higher priority than a new system for domains. Python 2 is hitting end of life at the end of the year, there's the postgres cutover and the full conversion of Attribute-based data storage to other models, the normalization of the user auth model, etc.

**Comment by smreyno** [comment]
Off-hand as a stop-gap I'd suggest: 
* Controlling for resource inflation in some respect, or maybe assessing resources differently if they're being used to enact permanent changes to domains. I do think there's a material difference in someone converting resources to silver to improve their gear and someone using them to make permanent changes to the IC feudal landscape, so it might be hard to do this without nerfing people's ability to use them for smaller-scale economic things, idk. 
* Related to the above, upping the difficulty of raising an org modifier via work past a certain level, maybe adjusted for what the org's IC power would logically be (like a Great House can grind to a higher level more easily than a March can). At present, org investing seems to be the primary route to tangible domain improvement, and once those numbers reach a certain point, it seems like it just doubles down on the issue of how inflated resources are. 
* A clarification on how much @armies matter/how they're being used to calculate a House's strength. Maybe a chart for what's 'normal' for Houses at certain ranks. At present, with how diffuse things are, it can feel like a waste to invest in them right up until the moment it's clearly not/they're necessary for big story moves.

These are just off the top of my head, idk how good they are or if they'd work well as controls.

**Comment by Zironic** [comment]
I took a stab at starting a design specification for a complete Domain system. Let me know if you like the direction it is taking since writing a complete specification is a lot of work and there's little reason to do it if you don't want it!

https://docs.google.com/document/d/1v1yA1W-lfyKUtO6ffEhhvVcfbx_K0rvkcyjO5LITSPI/edit?usp=sharing

**Comment by TehomCD** [comment]
Here's some of the design docs for Projects/Dominion: it has a pretty different vibe than a simple 4X minigame. https://docs.google.com/document/d/10jJ8f0jJBBJ1sRuqtak8CoDtg-KBS9sVWI-aEvUDVeY/edit#heading=h.ygfhxj2bugvq

**Comment by Zironic** [comment]
That project proposal confuses me because on one hand it has the goal of removing the need of GMs to perform domain management but on the other hand it seems under that proposal project completion will still be under staff review?

**Comment by TehomCD** [comment]
My understanding is that tasks would take from a shared pool with actions while being much more automated. So think of it as a subsystem for actions specific to domains. But the overall goal is to tie any sort of domain growth/management minigames into providing story hooks, either from things required for advancement or for various crises that'd be triggered by activity.

**Comment by TehomCD** [comment]
I also should mention that scarcity and loss is something I feel has to be an integral part of the system. Unfettered growth is a problem in that it eliminates any sort of RP around scarcity and divorces the world from concern about basic needs. I would very much want population growth to be slow and finite, and for any troop levies to draw upon a limited pool, for example.

The problem with work and unfettered inflation is probably something that needs to be solved in parallel - most likely removing resources from being allowed to directly purchase anything and instead be exchanged for silver in ways that are bottlenecked at some rate over time.

**Comment by Zironic** [comment]
Yeah, that's why I focused my proposal on making sure that the core mechanics of the Domain system are self-limiting by making an explicit link between the Domain population and the military in addition to limiting the military size based on military infrastructure. That way the mechanics of Dominion itself ensures you have an idea for how powerful any given domain is going to be and there's a limit to how much you can compensate by just pouring more money/resources into the domain.

I think a core component of designing a Domain system that is reasonably stable is to prevent additional Player Characters to provide linear scaling. The main issues are

1. Every new Player Character generates a very significant amount of resources every week
2. Every new Player Character brings with them 2 new Action Slots.

I think the way around that is to first give Domains their own resource that players can't generate through any other means and I think the way around the second is to give Domains their own action pool so that while players can assist the Domain in the Domain projects, having additional PCs doesn't mean you can do more Domain projects simultaneously.

**Comment by dreamsofgold** [comment]
I don't think domains should have their own resources, but that improvements based on scale would accordingly require a greater amount of investments, just as resource modifiers do, at present. But perhaps make it something time-locked like RFRs and gradual to give a sense of hard-won growth, with indicators to go with it.

OTOH, because I am generally against grandfathering policies made by either rule or code, I'd suggest those domains beneath a certain threshold to be able to improve more, up to the point they've had a more or less fair 'catch up' with others. 

Armies are some of the least abstract resources on the game, with a fixed cost to recruit AND to maintain. What I do suggest is that armies get taxed directly on the income rather than the bank account.

**Comment by TehomCD** [comment]
I'm not sure if a new resource needs to be anything other than time. Nothing should be instant, and we could bottleneck how quickly any number of things can be improved/upgraded simultaneously and how long it takes, and allow for things to fail/have setbacks.

I'd prefer having multiple bottlenecks: time, monetary investment, and character effort. It's much friendlier to remove bottlenecks later than to add them in, and any given bottleneck provides some possible lever to use for RP around scarcity/solving it.

**Comment by dreamsofgold** [comment]
Bottle-necking and limiting the amount of things you can improve at a time will work as far as a design choice goes. Plus, scaling difficulty according to the size of the domain adds a level of challenge that people will have incentive to rise to the occasion to.
